### ISSUE TITLE
Reset Component that can be tagged on to entities to reset once similar to Hidden in Amethyst

### ISSUE THREAD

**Issue by Skytrias** [Skytrias/everpuzzle#35]
This would be pretty nice to have since right now you have to manually get all WriteStorages and call their reset function, resulting in a lot of systemdata. Instead we could just add on a "Reset" Component to all Entities that should be reset without having to refer to them (alright atleast point at their entity). 

Here is some Info on the hidden struct, maybe take a look at the source code too.
https://www.amethyst.rs/doc/master/doc/amethyst_renderer/struct.Hidden.html

**Comment by monospacedmagic** [comment]
I'd like to tackle this one! 😃

**Comment by Skytrias** [comment]
@monospacedmagic as an example https://github.com/Skytrias/everpuzzle/blob/master/src/systems/cursor_system.rs#L140

i call a lot of defaults here and what could be done is something similar to the Hidden Component i described earlier.

basically if you want to hide a sprite all you need to do is add a Hidden Component to the entity https://github.com/Skytrias/everpuzzle/blob/master/src/systems/block_system.rs#L103 so you push it only once and once you dont use it anymore you take it off of the entity by removing it

So we could try adding a reset component similar to this that calls the enitities sub compoenents::default() funcition. i dont know if this even works but you can experiment

**Comment by monospacedmagic** [comment]
I see, thanks a lot for explaining! I'll start working on this probably on Sunday evening if that's okay.
